apps/docs/src/guide/provider/index.md

```diff
@@ -0,0 +1,48 @@
+# Provider
+
+The [`Provider`](../../api/Account/Provider.md) class lets you connect to a Fuel node ([_*local*_](../introduction/getting-started.md#connecting-to-a-local-node) or [_*external*_](../introduction/getting-started.md#connecting-to-the-testnet)) and interact with it, encapsulating common client operations in the SDK.
+
+<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#provider-definition{ts:line-numbers}
+
+## Provider Options
```

Comment:
I did not describe `cacheUtxo` here because I couldn't figure out how to convey why it'd be important to the end user.  #1515 questions the need for this functionality as well. Also, because it's an opt-in feature, I doubt that our users were even using it to begin with, and I don't remember any reports in `beta-4` and `beta-5` being made for the issue this caching solved. Perhaps we don't need it anymore.

If somebody can provide a user-relevant explanation for it, I'd appreciate it very much. If not, we can handle this part of the documentation alongside #1515 as well.

Reply:
If the flag is there, we should document it.

If later it is removed, docs can also removed with it.

Reply:
@Torres-ssf Objections?

Reply:
@arboleya I do agree that we should document this, but for now, I am not certain why this was added.

I will take a look into this and add some relevant explanations to our docs.

Reply:
https://github.com/FuelLabs/fuels-ts/issues/2056